### PR TITLE
All spear throwing enhancement and fix (with 3 types of throwing mechanic) (broken currently)

### PR DESCRIPTION
**Pull request by Aceship**
- Moved spear projectile spawn to hand instead of the spear position (more accurate throwing and fix chitin spear projectile)
- Change spear model rotation to follow the direction it gonna throw when holding grab (better information of where it go)

### CODE REVIEW

File: ValheimVRMod/Scripts/SpearManager.cs [hunk 1]
```diff
@@ -22,22 +24,37 @@ public class SpearManager : MonoBehaviour {
         public static Vector3 aimDir;
         public static Vector3 startAim; 
         public static bool isThrowing;
+        private GameObject rotSave;
+        private static bool isPressed;
```
Comment (brandonmousseau) [inline ValheimVRMod/Scripts/SpearManager.cs:28]: what is "isPressed" referring to ?

Comment (Aceship) [reply ValheimVRMod/Scripts/SpearManager.cs:28]: when the grab button is pressed

Comment (Aceship) [reply ValheimVRMod/Scripts/SpearManager.cs]: changed it to `isThrowingStance` on newest push, to make it easier to know what its for 👍

**Comment by Aceship** [comment]
Not used anymore

**Pull request closed without merging**